src/System.Web.OData/OData/Query/Expressions/DynamicTypeWrapper.cs

```diff
@@ -120,4 +124,67 @@ internal class AggregationWrapper : GroupByWrapper
     internal class NoGroupByAggregationWrapper : GroupByWrapper
     {
     }
+
+    internal class ComputeWrapper<T> : GroupByWrapper, IEdmEntityObject
+    {
+        public T Instance { get; set; }
+
+        public IEdmModel Model { get; set; }
+
+        private bool _merged;
+        protected override void EnsureValues()
+        {
+            base.EnsureValues();
+            if (!this._merged)
+            {
+                // Base properties availbale via Instance can be real OData properties or generated in previous transformations
+
+                var instanceContainer = this.Instance as DynamicTypeWrapper;
+                if (instanceContainer != null)
+                {
+                    // Add proeprties generated in previous transformations to the collection
+                    _values = _values.Concat(instanceContainer.Values).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+                }
+                else
+                {
+                    // Add real OData properties to the collection
+                    // We need to use injected Model to real proeprty names
+                    var edmType = GetEdmType() as IEdmEntityTypeReference;
+                    _typedEdmEntityObject = _typedEdmEntityObject ??
+                        new TypedEdmEntityObject(Instance, edmType, GetModel());
+
+                    var props = edmType.DeclaredStructuralProperties().Where(p => p.Type.IsPrimitive()).Select(p => p.Name);
+                    foreach (var propertyName in props)
+                    {
+                        object value;
+                        if (_typedEdmEntityObject.TryGetPropertyValue(propertyName, out value))
+                        {
+                            _values.Add(propertyName, value);
```

Comment:
>Add [](start = 36, length = 3)

If there is a conflict with names this will fail. Consider using set [].

Reply:
In your case performance is the same so the only question is do you expect conflicts?

---
In reply to: [172286870](https://github.com/kosinsky/WebApi/pull/22#discussion_r172286870) [](ancestors = 172286870)